Shared/UserAgent.swift

```diff
@@ -28,41 +28,110 @@ open class UserAgent {
         return clientUserAgent(prefix: "Firefox-iOS")
     }
 
-    public static func defaultUserAgent() -> String {
+    public static func isDesktop(ua: String) -> Bool {
+        return ua.lowercased().contains("intel mac")
+    }
+
+    public static func desktopUserAgent() -> String {
+        return UserAgentBuilder.defaultDesktopUserAgent().userAgent()
+    }
+
+    public static func mobileUserAgent() -> String {
+        return UserAgentBuilder.defaultMobileUserAgent().userAgent()
+    }
+
+    public static func oppositeUserAgent(domain: String) -> String {
+        let isDefaultUADesktop = UserAgent.isDesktop(ua: UserAgent.getUserAgent(domain: domain))
+        if isDefaultUADesktop {
+            return UserAgent.getUserAgent(domain: domain, platform: .Mobile)
+        } else {
+            return UserAgent.getUserAgent(domain: domain, platform: .Desktop)
+        }
+    }
+
+    public static func getUserAgent(domain: String, platform: UserAgentPlatform) -> String {
+        switch platform {
+        case .Desktop:
+            if let customUA = CustomUserAgentConstant.desktopUserAgent[domain] {
+                return customUA
+            } else {
+                return desktopUserAgent()
+            }
+        case .Mobile:
+            if let customUA = CustomUserAgentConstant.mobileUserAgent[domain] {
+                return customUA
+            } else {
+                return mobileUserAgent()
+            }
+        }
+    }
+
+    public static func getUserAgent(domain: String = "") -> String {
         // As of iOS 13 using a hidden webview method does not return the correct UA on
         // iPad (it returns mobile UA). We should consider that method no longer reliable.
         if UIDevice.current.userInterfaceIdiom == .pad {
-            return desktopUserAgent()
+            return getUserAgent(domain: domain, platform: .Desktop)
         } else {
-            return mobileUserAgent()
+            return getUserAgent(domain: domain, platform: .Mobile)
         }
     }
 
-    public static func isDesktop(ua: String) -> Bool {
-        return ua.lowercased().contains("intel mac")
+}
+
+public enum UserAgentPlatform {
+    case Desktop
+    case Mobile
+}
+
+public struct CustomUserAgentConstant {
+    private static let defaultMobileUA = UserAgentBuilder.defaultMobileUserAgent().userAgent()
+    private static let customDesktopUA = UserAgentBuilder.defaultDesktopUserAgent().clone(extensions: "Version/\(AppInfo.appVersion) \(UserAgent.uaBitSafari)")
```

Comment:
This should probably use fixed Firefox version (21)

Reply:
Fixed.